tools/integration_tests/emulator_tests/proxy_server/operation_manager_test.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proxy_server
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewOperationManager(t *testing.T) {
+	config := Config{
+		RetryConfig: []RetryConfig{
+			{Method: "JsonCreate", RetryInstruction: "return-503", RetryCount: 2, SkipCount: 1},
+			{Method: "JsonStat", RetryInstruction: "stall-33s-after-20K", RetryCount: 3, SkipCount: 0},
+		},
+	}
+
+	om := NewOperationManager(config)
+
+	// Assert that retryConfigs are initialized correctly
+	assert.Len(t, om.retryConfigs, 2)
+	assert.Len(t, om.retryConfigs["JsonCreate"], 1)
+	assert.Len(t, om.retryConfigs["JsonStat"], 1)
+
+	assert.Equal(t, "return-503", om.retryConfigs["JsonCreate"][0].RetryInstruction)
+	assert.Equal(t, "stall-33s-after-20K", om.retryConfigs["JsonStat"][0].RetryInstruction)
+}
+
+func TestRetrieveOperation(t *testing.T) {
```

Comment:
Can we add a test?
- with two retry configs (both same request type, both for different request type)

Reply:
Done